autoPyTorch/data/tabular_feature_validator.py

```diff
@@ -478,33 +495,87 @@ def impute_nan_in_categories(self,
         # TypeError: '<' not supported between instances of 'int' and 'str'
         # in the encoding
         for column in self.enc_columns:
-            if X[column].isna().any():
-                if column not in self.dict_missing_value_per_col:
-                    try:
-                        float(X[column].dropna().values[0])
-                        can_cast_as_number = True
-                    except Exception:
-                        can_cast_as_number = False
-                    if can_cast_as_number:
-                        # In this case, we expect to have a number as category
-                        # it might be string, but its value represent a number
-                        missing_value: Union[str, int] = '-1' if isinstance(X[column].dropna().values[0], str) else -1
-                    else:
-                        missing_value = 'Missing!'
-
-                    # Make sure this missing value is not seen before
-                    # Do this check for categorical columns
-                    # else modify the value
-                    if hasattr(X[column], 'cat'):
-                        while missing_value in X[column].cat.categories:
-                            if isinstance(missing_value, str):
-                                missing_value += '0'
-                            else:
-                                missing_value += missing_value
-                    self.dict_missing_value_per_col[column] = missing_value
+            # no missing values for categorical column
+            if not X[column].isna().any():
+                continue
+
+            if column not in self.dict_missing_value_per_col:
+
+                first_value = X[column].dropna().values[0]
+
+                if can_cast_as_number(first_value):
```

Comment:
This is an additional edge case. The whole idea is for values to be either ints represented as string, or strings, this would be a mix of both. Going through the unique values and seeing if they can all be converted to floats is a smart idea. 

On an additional note, all of these values will be encoded, so, although I think imputing -1 and a string is nice, I also wonder to what use, when in the end it will be an additional column at least for the case of one-hot encoding.